app/models/committees/agenda.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class Agenda < ApplicationRecord
+  # search
+  include PgSearch
+  pg_search_scope(
+    :search,
+    against: %i[description],
+    using: { tsearch: { prefix: true } }
+  )
+
+  # relations
+  belongs_to :unit
+  belongs_to :agenda_type
+
+  # validations
+  validates :description, presence: true
+  validates :status, presence: true
+
+  # enums
+  enum status: { newly: 0, desided: 1, delayed: 2 }
```

Comment:
`newly` => `new`
`desided` => `decided`

daha doğru.

Reply:
New Activerecord tarafından kullanıldığı için kullanmamıştım.

Reply:
Sınıf metodu olan `new` ile karışmaz, sınıf üzerinde kullanılmadığı sürece.  Ruby bu isim konularında çok esnektir, keyword gibi görünen (ki öyle değildir çoğu) neredeyse her identifier'ı kullanabilirsin.  Kullanmamak için tek geçerli neden yanıltıcılık olabilir ama burada geçen Rails enum zaten `new?` vb formda kullanılacağından yanıltıcı da olmuyor.  En uygun isim bu mudur bilmiyorum ama gönül rahatlığıyla kullanabilirsin.  Enum'lar okunurluğu çok arttırıyor, aman bunların isimlerini doğru seçin.

Reply:
> Enum'lar okunurluğu çok arttırıyor, aman bunların isimlerini doğru seçin.

Bakın mesela [API](https://api.rubyonrails.org/v5.2/classes/ActiveRecord/Enum.html) ne enstrümanlar sunuyor?  `prefix` ve `suffix` kullanımına imkan sunulmuş.  Bunlar çeşitli durumlarda hep okunurluğu arttıran **enstrümanlar**.  Duruma göre kullanabilirsiniz (ömrümüzün ciddi bir bölümü isimlendirmelerle geçiyor).  Ara ara API boş kaldıkça API dokümanlarının üzerinden geçelim lütfen.  Çok yararlı olacağına emin olabilirsiniz.

Reply:
> Sınıf metodu olan `new` ile karışmaz, sınıf üzerinde kullanılmadığı sürece.

Hocam sınıf üzerinde de kullanılabiliyor. Bende ilk önce `new` kullandım, hata aldığım için alternatif başka bir status eklemek zorunda kalmıştım.

https://stackoverflow.com/questions/22940415/enum-option-new-not-working